llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -4526,6 +4506,253 @@ static bool willGenerateVectors(VPlan &Plan, ElementCount VF,
   return false;
 }
 
+/// Get the VF scaling factor applied to the recipe's output, if the recipe has
```

Comment:
Nit, I presume nothing changed in calculateRegisterUsage? To make the diff smaller you could forward declare calculateRegisterUsage so you don't need to move it. And then you could move it in a follow up NFC.

Reply:
+1 to that :)

Reply:
Yep, this reminded me me that I was planning to move the code to VPlanAnalysis.cpp after this patch lands: https://github.com/llvm/llvm-project/pull/135673